src/command/mod.rs

```diff
@@ -16,24 +16,15 @@ pub enum State {
     Selected,
 }
 
-pub struct Command {
-    state: State,
-}
-
 const SCROLL_FACTOR: usize = 2;
 
-impl Command {
-    pub fn new() -> Command {
-        Command {
-            state: State::Insert,
-        }
-    }
-
-    pub fn treat_event<T>(&mut self, content: &mut T, view: &mut View, event: Event) -> bool
+impl State {
```

Comment:
This breaks separation of concerns. `State` ought to only contain proper state operations, and certainly not the command state machine.